### ISSUE TITLE
Invalid Syntax Error installing pylint on MacOs

### ISSUE THREAD

**Issue by demisx** [PyCQA/pylint#3824]
I have both python v2 (came with MacOs) and python v3 installed.

```sh
$ which python
/usr/bin/python
$ python --version
Python 2.7.16

$ which phytho3
/usr/bin/python3
$ python3 --version
Python 3.7.3
```

### Steps to reproduce
1. Attempt to install pylint with pip v3
```
pip3 install pylint
```

### Current behavior
```
Collecting pylint
  Using cached pylint-2.6.0-py3-none-any.whl (325 kB)
Collecting isort<6,>=4.2.5
  Using cached isort-5.5.1-py3-none-any.whl (95 kB)
Collecting mccabe<0.7,>=0.6
  Using cached mccabe-0.6.1-py2.py3-none-any.whl (8.6 kB)
Collecting astroid<=2.5,>=2.4.0
  Using cached astroid-2.4.2-py3-none-any.whl (213 kB)
Collecting toml>=0.7.1
  Using cached toml-0.10.1-py2.py3-none-any.whl (19 kB)
Collecting lazy-object-proxy==1.4.*
  Using cached lazy-object-proxy-1.4.3.tar.gz (34 kB)
  Installing build dependencies ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/local/opt/python@3.8/bin/python3.8 /usr/local/lib/python3.8/site-packages/pip install --ignore-installed --no-user --prefix /private/var/folders/yn/tfydvv613h95b9364681z6k00000gn/T/pip-build-env-ji0t120p/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'setuptools>=30.3.0' wheel 'setuptools_scm>=3.3.1'
       cwd: None
  Complete output (31 lines):
  Traceback (most recent call last):
    File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 194, in _run_module_as_main
      return _run_code(code, main_globals, None,
    File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 87, in _run_code
      exec(code, run_globals)
    File "/usr/local/lib/python3.8/site-packages/pip/__main__.py", line 26, in <module>
      sys.exit(_main())
    File "/usr/local/lib/python3.8/site-packages/pip/_internal/cli/main.py", line 73, in main
      command = create_command(cmd_name, isolated=("--isolated" in cmd_args))
    File "/usr/local/lib/python3.8/site-packages/pip/_internal/commands/__init__.py", line 104, in create_command
      module = importlib.import_module(module_path)
    File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/importlib/__init__.py", line 127, in import_module
      return _bootstrap._gcd_import(name[level:], package, level)
    File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
    File "<frozen importlib._bootstrap>", line 991, in _find_and_load
    File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
    File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
    File "<frozen importlib._bootstrap_external>", line 783, in exec_module
    File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
    File "/usr/local/lib/python3.8/site-packages/pip/_internal/commands/install.py", line 17, in <module>
      from pip._internal.cli.req_command import RequirementCommand, with_cleanup
    File "/usr/local/lib/python3.8/site-packages/pip/_internal/cli/req_command.py", line 22, in <module>
      from pip._internal.req.constructors import (
    File "/usr/local/lib/python3.8/site-packages/pip/_internal/req/__init__.py", line 10, in <module>
      from .req_install import InstallRequirement
    File "/usr/local/lib/python3.8/site-packages/pip/_internal/req/req_install.py", line 10, in <module>
      import uuid
    File "/usr/local/lib/python3.8/site-packages/uuid.py", line 138
      if not 0 <= time_low < 1<<32L:
                                  ^
  SyntaxError: invalid syntax
  ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/local/opt/python@3.8/bin/python3.8 /usr/local/lib/python3.8/site-packages/pip install --ignore-installed --no-user --prefix /private/var/folders/yn/tfydvv613h95b9364681z6k00000gn/T/pip-build-env-ji0t120p/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'setuptools>=30.3.0' wheel 'setuptools_scm>=3.3.1' Check the logs for full command output.
```

### Expected behavior
No error with pylint v3 installed

### pylint --version output
```
$ pylint --version
Failed to execute process '/usr/local/bin/pylint'. Reason:
The file '/usr/local/bin/pylint' specified the interpreter '/usr/local/opt/python/bin/python3.7', which is not an executable command.
```

**Comment by AWhetter** [comment]
It looks like you have a Python 2 only version of the uuid module, which is usually part of the standard library, in your site-packages directory. Also your Python 3 executable says it's Python 3.7 but it looks like pip3 is using Python 3.8.
You may want to clean up your Python installations and start again. I don't know much about Mac specifically but I would recommend following either https://docs.python-guide.org/starting/install3/osx/ or using [pyenv](https://github.com/pyenv/pyenv).

**Comment by demisx** [comment]
Thank you. Something was indeed messed up in python installation, because there are so many of them (MacOs default, XCode, Brew). I've reinstalled XCode and followed the linked above article. It worked. I also prefer `docker` to `pyenv`.
